Flask add

jakeraffe opened the pull request:
Created flask skeleton for backend. Basic Hello world application with barebones flask. No implementation for API as of now, but it will come soon. README.txt explaines how to run flask app with the vm.

Commit:
backend-flask

Second attempt

jakeraffe committed:
Update routes.py

BluuArc commented on backend-flask/app/routes.py:
```diff
@@ -0,0 +1,15 @@
+ from flask import render_template, jsonify
```
Got an indent error here

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,25 @@
+$ source venv/bin/activate
+
+For windows users using cmd:
+$ venv\Scripts\activate
```
Just a heads up that the `Activate.ps1` script didn't work for me on Powershell due to the following error.

```
.\venv\Scripts\Activate.ps1 : File .\LUG_Dungeons_and_Dragons_Webbapp\backend-flask\venv\Scripts\Activate.ps1 cannot be loaded because running scripts is disabled on this system. For more information, see
about_Execution_Policies at https:/go.microsoft.com/fwlink/?LinkID=135170.
```

However, running `activate.bat` in the regular Windows command prompt (i.e. `cmd`) worked fine.

wilt00 replied:
Might be a good idea to add in a note about setting your Powershell execution policy settings (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-5.1)

jakeraffe replied:
Added a note in the Readme.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,25 @@
+
+$ python3 -m venv venv
+
+This installs the venv package and names the folder "venv".
```
If you don't want people to commit their `venv` folders, you can create a `.gitignore` file and add the folder to it. VSCode gives some helpful highlighting in its folder tree to indicate what is ignored when committing files; I do not know if Atom has a similar feature, but it likely does.

jakeraffe replied:
Done. I just tested it and the .gitignore ignores the venv folder.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,25 @@
+DEBUGGING
+If the debugger does not activate while testing app, use the following flag to constantly update page without needing to restart the server:
+
+$ export FLASK_DEBUG=1
```
You don't have any instructions on actually running the server. I got it working by running `flask run` in the `backend-flask` folder.

Additionally, the Windows `cmd` command for setting the `FLASK_DEBUG` variable is `set FLASK_DEBUG=1`

BluuArc reviewed (changes requested):
Nice job on creating the initial setup. I made some comments mostly about documentation and one indent error.

One more thing I wasn't sure of is if the `.pyc` files should be committed to the repository (and whether or not they should be added to a `.gitignore`). Hopefully @Benbigmac, @Reisande, or @BreadYogurt can answer that.

wilt00 commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,25 @@
+VENV
+To install venv enter the following command:
+
+$ python3 -m venv venv
```
If we're using virtualenvs, then we should also have a requirements.txt file checked in to the repository with the specific version of Flask we're using, and probably also the linter we're using (pylint?).

Benbigmac replied:
what were we using for the website? do we need virtual envs? Just asking because I know the flask tutorials seem to have a thing for them like a Texan has a thing for barbecues

jakeraffe replied:
Will, I have pylint installed and was just using it as a matter of fact. I think its the best linter from my few minutes of research. 
Also Ben, I agree with you. The only reason why I continued the trend was because I saw this post and the person brings up a good point.
https://stackoverflow.com/questions/32756711/is-it-necessary-to-use-virtualenv-to-use-flask-framework

Benbigmac replied:
I was talking with people and venv is apparently necessary

Commit:
spelled gitignore wrong

Commit:
.gitgore, README.txt, and requiremtnts.txt update

First of all, I changed the structure of the folder to me more centralized by getting rid of the __init__.py file. I Added a .gitgore file for the pycache folder and venv.  I also added a requirements.txt file wich lets users quickly install flask and other required packages. Additionally, I started a config file which will give support to the RESTful API.

Commit:
update requirement.txt

Added pylint installation to requirements file.

Commit:
Resolved README issues

Added a note about powershell issues with running. The app.

BluuArc commented on backend-flask/.gitignore:
```diff
@@ -0,0 +1,15 @@
+# Byte-code files
+__pycache__/
+
```
Should also add `*.pyc` to this. Testing out the instructions in the readme led to `app.pyc` being created.

jakeraffe replied:
Thats what the __pycache__/ is for in the gitignore, since .pyc files fall within that folder. Ill add your code too just to be sure it doesnt make it in there.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,70 @@
+$ source venv/bin/activate
+
+For windows users using cmd:
+> venv\Scripts\activate
```
Specifically, it's `activate.bat`

jakeraffe replied:
Added.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,70 @@
+For example, if our application was named DND.py we would input the following flag in the terminal:
+
+For Linux:
+$ export FLASK_APP=DND.py
```
Testing out the instructions with `DND.py` did not work, but using `app.py` works.

jakeraffe replied:
Yea I changed the file to app.py so this is done. Also changed in README as well now.

BluuArc replied:
As long as you update the rest of the instructions to reflect this change, then it's fine.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,70 @@
+To actually run the application, input the following:
+$ flask run
+    or
+$ python DND.py
```
Not sure if it's just my config, but running `python DND.py` or `python app.py` did not work for me. I can confirm that `flask run` does work.

jakeraffe replied:
From the docs: 
"from Flask 0.11 onward the flask command is recommended. The reason for this is that due to how the dev server's reload mechanism works there are some bizarre side-effects when using app.run (like executing certain code twice, sometimes crashing without message or dying when a syntax or import error happens)."

If "flask run" is the only command that works its fine because its what they recommend for the above reasons.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,70 @@
+VENV
+To install venv enter the following command:
+
+$ python3 -m venv venv
```
(Minor) Inconsistent use of the command used. Here, you used `python3`, while in the Running the App section you used just `python`. Make sure it's consistent.

jakeraffe replied:
You have to specify a version, 'python3', when installing venv otherwise you will be met with an error:

"$ python -m venv venv
/usr/bin/python: No module named venv"

While running a python.py file you dont need to specify version so the '3' is not required. You can but you dont need to. Honestly I am going to take out the "python app.py" run command because it is deprecated.

Commit:
Documentation updates

Updated and added extra .gitignore attribute to ignore *pyc files, changed terminology in README and removed the deprecated "python app.py" command.

jakeraffe commented:
Updated and satisfied all requested changes.

BluuArc commented on backend-flask/README.txt:
```diff
@@ -0,0 +1,68 @@
+VENV
+To install venv enter the following command:
+
+$ python3 -m venv venv
```
I would add a note here that when installing `venv`, use the version of python that support venv (i.e. 3.3+), as `python3` isn't consistent across all systems. For example, `py` refers to Python 3.7.1 for me while `python` refers to Python 2.x.

It's just a minor wording thing; up to you if you want to change it.

BluuArc reviewed (approved):
Just one minor comment on the specific usage of `python3`, but otherwise looks fine to me. Nice job!
